### ISSUE TITLE
Async filter values get lost after getRows (even with newRowsAction)

### ISSUE THREAD

**Issue by chrys1110** [ag-grid/ag-grid#2042]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Applying a "set"-filter triggers getRows in the enterprise-datasource. But the values get lost if they came from an async call. You cannot even keep it with the newRowsAction-prop.

**Expected behavior**
Applying a "set"-filter triggers getRows in the enterprise-datasource. And the values-function gets recalled. Or you can keep it with the newRowsAction-prop "keep".

**Minimal reproduction of the problem with instructions**
```
<AgGridReact

columnDefs={[
{ headerName: 'Type', field: 'type',
        filter: 'set',
        filterParams: {
            values: params => setTimeout(() => params.success(["Type1","Type2"]), 100),
            newRowsAction: 'keep'
        }
    },
]}

onGridReady={gridParams => {
   gridParams.api.setEnterpriseDatasource({
      getRows: datasourceParams => {
         myFetchDataFunction(datasourceParams.request, (rows, totalRows)=>
            datasourceParams.successCallback(rows, totalRows);
         );
      }
  })
}}

rowModelType="enterprise"
enableServerSideFilter
/>
```
* Grid loads data
* **Click on filter at the type-column**
* "Type1, Type2" get loaded as values
* ag-grid calls getRows again
* only selectable value left in filter is "(blanks)"

**What is the motivation / use case for changing the behavior?**
It's working as one would expect.

**Please tell us about your environment:**
I don't think the environment is responsible for this.

* **ag-Grid version:**
ag-grid 14.1.1
ag-grid-enterprise 14.1.1
ag-grid-react 14.0.0

* **Browser:**
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36

* **Language:** ES6

**Comment by makinggoodsoftware** [comment]
Hi,

I believe this should have been fixed in ag-grid 14.2

Can you please try there?

Thanks

===============

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.

Bugs Raised Here:

We will do a periodic review of GitHub issues for any bugs raised and these can be tracked on our pipeline page: https://www.ag-grid.com/ag-grid-pipeline

Feature Requests/Support Raised Here:

We are reviewing how we process Feature Requests that come from GitHub. We currently have a very full pipeline of work which will be our priority for the coming months. 

As a result, we don’t have capacity to take feature requests from GitHub.

We will review this as we move through our current backlog.

Thanks for your understanding.
ag-Grid Team.

**Comment by ahmadshakeel78** [comment]
hi ,
i don't think so its an issue.may be you are doing some silly mistakes like you also update you colDfs with new request that's why its refresh all the filter and sort values as well
